packages/smart-contracts/scripts-create2/contract-setup/setupBatchPayments.ts

```diff
@@ -23,24 +23,26 @@ export const setupBatchPayments = async (
   );
   await Promise.all(
     hre.config.xdeploy.networks.map(async (network) => {
-      let provider;
-      if (network === 'celo') {
-        provider = utils.getCeloProvider();
-      } else {
-        provider = utils.getDefaultProvider(network);
-      }
-      const wallet = new hre.ethers.Wallet(hre.config.xdeploy.signer, provider);
-      const signer = wallet.connect(provider);
-      const batchPaymentConnected = await batchPaymentContract.connect(signer);
-      const adminNonce = await signer.getTransactionCount();
-      const gasPrice = await provider.getGasPrice();
+      try {
```

Comment:
see next file

Reply:
We do not want to stop the execution. 
The admin tasks happen on all networks, so if it fails on one we don't want to prevent them to run on others.

If the admin task fails, it will most likely be due to missing funds. And a message is displayed in the console. 
The advantage is that in the event of any error, the person deploying the contracts just has to start the script again:
- Deployments already made will be skipped,
- Deployments that failed will be run,
- Admin tasks already performed will be skipped,
- Admin tasks that failed will be run

Reply:
+1 Léo
I got this issue and I added it myself to deploy the batch on every network easily.

Reply:
What if there is a miss config that will generate errors on all networks for all contracts? Won't this waste a lot of gas? In case of an error, can we stop and only save the artifacts for what's already been deployed, and then we try again after fixing the mistake:
- what's already been deployed will be skipped
- what's missing will be run and caught back

; do you think that's feasible?

Reply:
It won't waste gas since if the transaction is doomed to fail on all networks, it won't be fired because gas estimation won't work.
However it might lead to wasting gas sometimes, it should be small amounts since most chains are cheap aside mainnet.

The person deploying the contracts can stop the script anytime, save the artifacts, and restart the script when he wants to it will automatically catch up where he stopped.